### ISSUE TITLE
Ability to run backtest & get the results/outcomes of ALL buy==1.0 indicators

### ISSUE THREAD

**Issue by markdregan** [freqtrade/freqtrade#6270]
I've thoroughly read the documentation and asked questions on Discord. Greatly appreciate the feedback here.

**TLDR:** For all the buy signals generated by a strategy, I need to compute the outcome of each one of these trades as if the strategy were to have executed on it. In short, I need to simulate (or run backtest) each buy indicator that a strategy makes.

**Challenge:** When we run a freqtrade Backtest, only some of a pair's trades are simulated/run - because of the constraints of `max-open-trades` and other constraints of the strategy.

**What I've tried:** I've built a class that effectively runs many very small backtests. Each one I can ensure will return the results of the buy indicator I am interested in. Gathering all the results together, I have all the outcomes for all of the buy indicators a strategy produces. However, execution time is slow and some what brittle - so I'm not thinking I should double check there isn't an easier approach...

**Question:** Can the dev team advise if there is an easier way to achieve this goal? Or if there is an approach that you think would be worth perusing.

TY for any guidance on above. 

## Describe your environment

  * Operating system: Mac Monterey
  * Python Version: 3.8.9 (`python -V`)
  * CCXT version: 1.66.66 (`pip freeze | grep ccxt`)
  * Freqtrade Version: develop-fe735212 (`freqtrade -V` or `docker-compose run --rm freqtrade -V` for Freqtrade running in docker)

**Comment by stash86** [comment]
Setting max_open_trades to unlimited or a big number should help you?

**Comment by markdregan** [comment]
I believe that would still prevent multiple simultaneous positions on the same pair.

**Comment by stash86** [comment]
You can do so using the new dca feature.

please note that you can only sell as a whole though. You can't do partial sell right now

**Comment by markdregan** [comment]
Unfortunately that will give me one outcome for multiple stacked (dca) trades. My specific requirement is to compute the individual outcome for each buy indicator. And this is a hard requirement for the model I'm building.

Thanks for these ideas btw.

**Comment by stash86** [comment]
There was an old fork of freqtrade that can be used to manually buy coins, even those that already open, and sell it individually. But sadly it was not an automatic buy and currently has been retired in favor of the new dca feature.

it would have suit your need

**Comment by xmatthias** [comment]
You can use `--eps` for this - which will enter a new trade for buy signal (once per candle) (consider it deprecated, so no guarantees for how long this will stay around).

i'm not sure it'll play too well with DCA callbacks (and you cannot use unlimited stake, and need to start with a huge backtesting wallet to avoid running out of funds).
Also, it's a mode that's clearly NOT repeatable during anything other than backtesting, so once you got these results, you will need to filter down these results to reach similar results without `--eps`.

**Comment by markdregan** [comment]
Thank you @xmatthias and @stash86. This works very well.

Understood I should consider this deprecated. Allow me to explain the use case for your consideration when deprecating.

The use case is to generate a dataset with all buy triggers for a strategy + the signals/indicators available on that candle + the eventual outcomes each of these buy triggers (if it were made - ignoring constraints of wallet / max open orders etc).

This dataset if used for training a ML model to better understand when the strategy (primary rule model) is performing well or poorly. This secondary model is a 'critic' of the primary rule model - and is used to determine if we should be making trades from the strategy.

In summary - getting the outcomes of all buy triggers is a critical part of this workflow. The `--eps` and `--dmmp` technique works very well for obtaining trade outcomes for ALL buy triggers (not just the ones that could have been made given the constraints)

Much appreciation for all the work. TY.

**Comment by xmatthias** [comment]
Honestly - this usecase (get information for all buy signals)  is why we have this there - and why it's still there.

The problem is - it's often missunderstood - and many people assume that they'll be able to get to the same results in real trading.
I understand that you did get that part (it's used for research and elimination, not as "i want this result on my account") - unfortunately that's not how most people understand it.

While i do consider it deprecated - that doesn't necessarily mean we'll remove it short-term (just not improve it further) ... but i think it's okish as is.

**Comment by markdregan** [comment]
👍 I can appreciate that area of confusion. Depending on the knowledge level of the user - these more advanced flags/features have higher change of being mis-understood. Thanks again 👊
